cardinal/component/component_action.go

```diff
@@ -15,6 +15,38 @@ type Action struct {
 	AffectedByActionID uint32           `json:"affected_by_action_id"`
 }
 
+type ActionStore struct {
+	actions map[uint32]Action
+	nextID  uint32
+}
+
+func (ActionStore) Name() string {
```

Comment:
do we really need this? as in we can get this from a type assertion, something like if v, ok := interface{}(instance).(MyStruct); ok

Reply:
The thing is that is how the documentation from WE regarding cardinal components has. If that is missing, we will get the following problem
`[{"message": "\"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/component\".Action does not satisfy \"pkg.world.dev/world-engine/cardinal/types\".Component (missing method Name)",}]`
Therefore, it is needed for all components